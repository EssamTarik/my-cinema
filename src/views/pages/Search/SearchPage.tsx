import React from 'react';
import { MovieSearchResultsContainer, SearchBarContainer } from '../../../containers';
import './styles.css';

const SearchPage = () => (
  <div className='search-page'>
    <div className='search-page__search-bar-container'>
      <SearchBarContainer />
    </div>
    <div className='search-page__search-results-container'>
      <MovieSearchResultsContainer />
    </div>
  </div>
);

export default SearchPage;