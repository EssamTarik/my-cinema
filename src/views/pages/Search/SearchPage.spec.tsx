import React from "react";
import { render, cleanup } from "@testing-library/react";

import SearchPage from "./SearchPage";

jest.mock("../../../containers", () => ({
  MovieSearchResultsContainer: () => <div className="movie-search-results" />,
}));

jest.mock("../../../components", () => ({
  SearchBar: () => <div className="search-bar" />,
}));
const mockMovieSearchResults = jest.requireMock("movieSearchResults");

describe("SearchPage", () => {
  afterEach(cleanup);
  it("renders", () => {
    const { asFragment } = render(
      <SearchPage
        resetMovieSearch={jest.fn()}
        movieSearchState={mockMovieSearchResults}
        searchMovies={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
