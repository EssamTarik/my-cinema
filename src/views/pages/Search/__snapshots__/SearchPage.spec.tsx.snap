// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage renders 1`] = `
<DocumentFragment>
  <div
    class="search-page"
  >
    <div
      class="search-page__search-bar-container"
    >
      <div
        class="search-bar"
      />
    </div>
    <div
      class="search-page__search-results-container"
    >
      <div
        class="movie-search-results"
      />
    </div>
  </div>
</DocumentFragment>
`;
